Content.Server/GameObjects/Components/NodeContainer/Nodes/NodeFactory.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Content.Server.GameObjects.Components.NodeContainer.NodeGroups;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Reflection;
+using Robust.Shared.IoC;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Content.Server.GameObjects.Components.NodeContainer.Nodes
+{
+    public class NodeFactory
```

Comment:
Since you route this through IoC, I think it's a good idea to

  1. Make a `INodeFactory` interface, for mocking
  2. Slap doc comments on that interface